### PR TITLE
RNR-16: Add search to InviteFriends and AddScryers screens.

### PR DESCRIPTION
**Pull request by dblanchard13**
- https://scrychat.atlassian.net/browse/RNR-16
- TODO: Style the fucking search bar

**AddScryers with basic search bar**
simulator screen shot - iphone se - 2018-10-24 at 00 58 59

<hr>

**InviteFriends with basic search bar**
simulator screen shot - iphone se - 2018-10-24 at 00 59 09

### CODE REVIEW

File: src/screens/AddScryers.js [hunk 1]
```diff
@@ -90,12 +107,30 @@ class AddScryers extends Component {
...
+                  side, but want this attempt to be documented either way.
+
+                  clearIcon
+                  onClearText={ query => this.filterContacts(query) }
+                  ref={ search => this.search = search }
+                */}
+              <SearchBar
+                round
+                lightTheme
+                onChangeText={ query => this.filterContacts(query) }
```
Comment (jack829) [inline src/screens/AddScryers.js]: Let's put a slight delay on the search, like a debounce with 300ms, so the screen isn't just flashing around with every character

File: src/screens/InviteFriends.js [hunk 2]
```diff
@@ -81,16 +98,24 @@ class InviteFriends extends Component {
...
-            data={ contacts }
-            renderItem={ this.renderContact }
-            keyExtractor={ ({ id }) => id }
-          />
+        { allContacts.length ? (
+          <View>
+            <SearchBar
+              round
+              lightTheme
+              onChangeText={ query => this.filterContacts(query) }
```
Comment (jack829) [inline src/screens/InviteFriends.js]: i'd say debounce here too

Comment (dblanchard13) [reply src/screens/InviteFriends.js]: Good call 🙆‍♂️

**Comment by jack829** [comment]
Oh in the same vein of your comment ealier - would you mind just throwing a brief description of the branch as part of the branch name? Like `RNR-16-search-functionality` Just more explicitness

**Comment by dblanchard13** [comment]
Will do going forward, yeah

File: src/screens/AddScryers.js [hunk 3]
```diff
@@ -9,16 +9,19 @@ import {
 
 import AppText from '../components/AppText'
 import Icon from '../components/Icon'
+import { SearchBar } from 'react-native-elements'
 import { Actions } from 'react-native-router-flux'
 import Expo from 'expo'
 import API from '../API'
 import AuthInterface from '../AuthInterface'
 import { filterAndFormatIOSContactsInUnitedStates } from '../helpers/filterAndFormatIOSContactsInUnitedStates'
+import _ from 'lodash'
```
Comment (jack829) [inline src/screens/AddScryers.js]: lets change this to `import {debounce} from 'lodash'` and let's keep library imports at the top (we haven't been too strict on that but I think it helps...)

**Pull request merged**